File: Programacionavanzada/ejercicios/2243330341_Act2Clase27Feb_2/src/main/Practica03_b.java
package main;

import controller.AppService;
import controller.CategoriaController;
import model.StorageType;
import view.CategoriaInternalFrame;

import javax.swing.JDesktopPane;
import javax.swing.JFrame;
import javax.swing.SwingUtilities;

public class Practica03_b {
    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> {
            AppService service = new AppService();
            service.setStorageType(StorageType.TXT);

            JFrame frame = new JFrame("Parte1 - Practica03_b");
            frame.setSize(700, 470);
            frame.setLocationRelativeTo(null);
            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

            JDesktopPane desktopPane = new JDesktopPane();
            frame.setContentPane(desktopPane);

            CategoriaInternalFrame child = new CategoriaInternalFrame();
            desktopPane.add(child);
            new CategoriaController(child, service);
            child.setVisible(true);

            frame.setVisible(true);
        });
    }
}

File: Programacionavanzada/ejercicios/2243330341_ActClase27Feb_Practica2/src/2243330341_Practica2_python/main/practica03_b.py
import tkinter as tk

from PA_a2251330007_practica2_python.controller.app_service import AppService
from PA_a2251330007_practica2_python.controller.categoria_controller import CategoriaController
from PA_a2251330007_practica2_python.model.storage_type import StorageType
from PA_a2251330007_practica2_python.view.categoria_internal_frame import CategoriaInternalFrame


def main() -> None:
    service = AppService()
    service.set_storage_type(StorageType.TXT)

    root = tk.Tk()
    root.withdraw()
    child = CategoriaInternalFrame(root)
    child.title("Parte1 - Practica03_b")
    CategoriaController(child, service)
    child.set_on_close(root.destroy)
    root.mainloop()


if __name__ == "__main__":
    main()
